### PR TITLE
Add networktables for conecube, and calculate the rotation/translation vectors. 

### PR DESCRIPTION
**Pull request by cfinn68916**

### CODE REVIEW

File: SecondSight/Color.py [hunk 1]
```diff
@@ -27,173 +29,97 @@ def averageColor(frame, sampleRange):
...
+    for contour in cone_contours:
+        x, y, width, height, theta = cv2.minAreaRect(contour)
+        res.append(GamePiece(x, y, width, height, theta, True))
+    for contour in cube_contours:
+        x, y, width, height, theta = cv2.minAreaRect(contour)
+        res.append(GamePiece(x, y, width, height, theta, False))
+    return res
+
+
+cube_size = 10  # CM
```
Comment (joshbressers) [inline SecondSight/Color.py:62]: We need to make this code more general. Get the object name and size from configuration data. It was cubes this year, but we should expect red buoy in the water game next year

File: SecondSight/Color.py
Comment (joshbressers) [inline SecondSight/Color.py:1]: This needs tests created for the functionality. The test image I added has a blue cube shaped object in the frame, feel free to use it

File: SecondSight/main.py [hunk 2]
```diff
@@ -31,6 +30,7 @@ def mainLoop():
     # TODO: Every module should write to network tables, not from here
     networktables.NetworkTables.initialize(server=config.get_value('nt_dest'))
     april_table = networktables.NetworkTables.getTable('SecondSight').getSubTable('Apriltags')
+    conecube_table = networktables.NetworkTables.getTable('SecondSight').getSubTable('GamePieces')
```
Comment (joshbressers) [inline SecondSight/main.py:25]: I'm not keen on this being in main. Don't worry about it now, but we should think about how to do it better

File: SecondSight/__init__.py
Comment (joshbressers) [inline SecondSight/__init__.py:1]: I dig it!

**Review by joshbressers — CHANGES_REQUESTED** [review]
This looks great. We really do need to add tests though. It won't be hard to add them

File: SecondSight/main.py (hunk 2 above)
Comment (cfinn68916) [reply SecondSight/main.py:25]: It needs to be created somewhere, and fetching the table every time would be time-consuming. There is no better place to put it.

File: SecondSight/Color.py (hunk 1 above)
Comment (cfinn68916) [reply SecondSight/Color.py:62]: Let's not add a configuration value for the cube size, but we should add multiple functions for detecting different years game objects.

Comment (cfinn68916) [reply SecondSight/Color.py:62]: What we should do is have a configuration value that contains what should be detected. For example: ["apriltags2023", "cone2023", "ball2016"]. The code knows their shape and size, and can detect any of them.

Comment (joshbressers) [reply SecondSight/Color.py:62]: That makes sense. We should also allow for configuration arbitrary things. I could envision a universe where a demo bot is requested to track something by an observer. Being able to quickly add new objects to track would be extremely cool

Comment (cfinn68916) [reply SecondSight/Color.py:62]: That would be hard, but doable. The object could probably only be a few shapes, and needs to be uniformly colored. The good news is that the hard to detect game objects are often not lying around on the field, they are at human player stations(see 2019 hatch panels).

**Comment by joshbressers** [comment]
I'm going to approve this, but it still has a lot of issues. I think it will be easier to address them once this is merged.

In the future we should avoid such large PRs

**Review by joshbressers — APPROVED** [review]

**Pull request merged**